sklearn/model_selection/_split.py

```diff
@@ -798,24 +817,24 @@ class StratifiedGroupKFold(_BaseKFold):
     >>> X = np.ones((17, 2))
     >>> y = np.array([0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0])
     >>> groups = np.array([1, 1, 2, 2, 3, 3, 3, 4, 5, 5, 5, 5, 6, 6, 7, 8, 8])
-    >>> cv = StratifiedGroupKFold(n_splits=3)
-    >>> for train_idxs, test_idxs in cv.split(X, y, groups):
-    ...     print("TRAIN:", groups[train_idxs])
-    ...     print("      ", y[train_idxs])
-    ...     print(" TEST:", groups[test_idxs])
-    ...     print("      ", y[test_idxs])
-    TRAIN: [1 1 2 2 4 5 5 5 5 8 8]
-           [0 0 1 1 1 0 0 0 0 0 0]
-     TEST: [3 3 3 6 6 7]
-           [1 1 1 0 0 0]
-    TRAIN: [3 3 3 4 5 5 5 5 6 6 7]
-           [1 1 1 1 0 0 0 0 0 0 0]
-     TEST: [1 1 2 2 8 8]
-           [0 0 1 1 0 0]
-    TRAIN: [1 1 2 2 3 3 3 6 6 7 8 8]
-           [0 0 1 1 1 1 1 0 0 0 0 0]
-     TEST: [4 5 5 5 5]
-           [1 0 0 0 0]
+    >>> sgkf = StratifiedGroupKFold(n_splits=3)
+    >>> sgkf.get_n_splits(X, y)
+    3
+    >>> print(sgkf)
+    StratifiedGroupKFold(n_splits=3, random_state=None, shuffle=False)
+    >>> for i, (train_index, test_index) in enumerate(sgkf.split(X, y, groups)):
+    ...     print(f"Fold {i}:")
+    ...     print(f"  Train: index={train_index}, group={groups[train_index]}")
+    ...     print(f"  Test:  index={test_index}, group={groups[test_index]}")
+    Fold 0:
+      Train: index=[ 0  1  2  3  7  8  9 10 11 15 16], group=[1 1 2 2 4 5 5 5 5 8 8]
+      Test:  index=[ 4  5  6 12 13 14], group=[3 3 3 6 6 7]
+    Fold 1:
+      Train: index=[ 4  5  6  7  8  9 10 11 12 13 14], group=[3 3 3 4 5 5 5 5 6 6 7]
+      Test:  index=[ 0  1  2  3 15 16], group=[1 1 2 2 8 8]
+    Fold 2:
+      Train: index=[ 0  1  2  3  4  5  6 12 13 14 15 16], group=[1 1 2 2 3 3 3 6 6 7 8 8]
+      Test:  index=[ 7  8  9 10 11], group=[4 5 5 5 5]
```

Comment:
based on previous suggestion:
```suggestion
    Fold 0:
      Train: index=[ 0  1  2  3  7  8  9 10 11 15 16]
             group=[1 1 2 2 4 5 5 5 5 8 8]
      Test:  index=[ 4  5  6 12 13 14]
             group=[3 3 3 6 6 7]
    Fold 1:
      Train: index=[ 4  5  6  7  8  9 10 11 12 13 14]
             group=[3 3 3 4 5 5 5 5 6 6 7]
      Test:  index=[ 0  1  2  3 15 16]
             group=[1 1 2 2 8 8]
    Fold 2:
      Train: index=[ 0  1  2  3  4  5  6 12 13 14 15 16]
             group=[1 1 2 2 3 3 3 6 6 7 8 8]
      Test:  index=[ 7  8  9 10 11]
             group=[4 5 5 5 5]
```

Reply:
Thanks this is a good suggestion, I was too focused on consistency. Have amended to:
```python
    ...     print(f"Fold {i}:")
    ...     print(f"  Train: index={train_index}\n\t group={groups[train_index]}")
    ...     print(f"  Test:  index={test_index}\n\t group={groups[test_index]}")
```
WDYT? Result is same.

Reply:
good for me

Reply:
hum but it looks like the new line character is not valid there...

Reply:
Annoyingly, can't find a way around this, have just used your suggestion. Thanks!